corehq/apps/smsbillables/models.py

```diff
@@ -367,3 +399,51 @@ def _get_usage_fee(cls, domain, direction):
                 % (direction, domain)
             )
         return usage_fee
+
+    @classmethod
+    def _get_twilio_charges(cls, backend_message_id, direction, couch_id):
+        def _get_twilio_client():
+            twilio_backend = SQLTwilioBackend.get_global_backends_for_this_class(SQLTwilioBackend.SMS).get()
+            config = twilio_backend.config
+            return TwilioRestClient(config.account_sid, config.auth_token)
+
+        try:
+            if backend_message_id:
+                twilio_message = _get_twilio_client().messages.get(backend_message_id)
```

Comment:
see my previous comment about the possible timing issue here:

there's the possibility of a timing issue here since we'll create the billable task once we hand off the sms to twilio, but twilio might not process it by the time the billable task runs. so before we get the fee api response from twilio we should check the message status from twilio and if it's any of the ones i listed above we should retry the billable task later.